tests/test_directory_sync.py

```diff
@@ -283,3 +447,33 @@ def test_primary_email_none(self, mock_user_no_email, mock_request_method):
         me = primary_email.primary_email()
 
         assert me == None
+
+    def test_directories_auto_pagination(self, mock_directories, mock_request_method):
+        mock_request_method("get", mock_directories, 200)
+        directories = self.directory_sync.list_directories(limit=2)
```

Comment:
Should the `limit` be passed here? It seems like it's not being applied, given that the length of `all_directories` is larger than the limit in the check below. If the limit was being honored, I'd expect the length to be 2.

Reply:
The limit being passed here is to ensure that the initial list_directories call does not return the full list. It's not being used in the final assertion. 

That first list_directories call returns the first two values, and then the auto_paging_iter method returns the rest as all_directories. If all_directories matches the mock_directories list it pulled from then the test passes. 

I know there must be a better way to test this, but I do think the limit is being applied successfully. I'll look into programmatically creating a more robust mock list.

Reply:
The limit and other query params should be honored with the auto pagination. The auto pagination should only concern itself with paging through the list, but should flow through any parameters sent.